lib/r10k/puppetfile.rb

```diff
@@ -104,10 +104,10 @@ def load(default_branch_override = nil)
     if self.loaded?
       return @loaded_content
     else
-      if !File.readable?(@puppetfile_path)
```

Comment:
Previous to this a missing Puppetfile would cause this portion the deployment to be a noop, while a syntax error in your Puppetfile would still cause the run to fail, right?

It seems we may want to either fix the `@puppetfile_path` to be the same kind of path that the loader will use. Or catch SystemCallError (or whatever else a missing/unreadable file might throw) here?

Reply:
yes.

I have another PR I'm gonna put up that will change the path used here, but it seemed better to me to do the validation/rescuing in the method that would actually throw, especially since the `ModuleLoader` methods are the ones we actually plan to keep. Are you just thinking that, by putting it here, we'd avoid unnecessarily trying to load altogether? Or is there another reason you think that stuff should go here?

Reply:
I think previously the run would fail if their were syntax errors in the Puppetfile, while now those errors will be printed at debug and the run will pass. (I might be wrong about that, but if that's the case we shouldn't do it).

Good point about the exception handling, I was just spitballing ideas that would keep us from doing a blanket rescue here.

Reply:
yeah, you're right. I guess I wasn't really clear on _why_ we just log and move on if the Puppetfile is unreadable but throw if there's another kind of error - I assumed that we didn't want to fail the run if there was anything wrong with the Puppetfile in whatever case(s) `load` is used (as opposed to `load!`), but didn't investigate at all. I can bring back the old behavior if you think that's better. I guess a change would probably be considered breaking by _someone_ huh..

Reply:
So I don't think you need a Puppetfile in an environment if you either don't have dependencies or you vendor them all. In that case the load becomes a no-op and the run continues. However, the attempting to deploy external dependencies is an opt-in feature so maybe it should be an error if you opt-in to deploying modules but there is not Puppetfile? I suppose you could do a deploy all kinda thing and want it to deploy modules if the Puppetfile exists in an environment?

Reply:
I removed this change for now.